### ISSUE TITLE
ADAL angular logout is not working

### ISSUE THREAD

**Issue by neelDev** [AzureAD/azure-activedirectory-library-for-js#549]
Hello,

I am facing the problem with ADAL logout. Azure AD is not redirecting to logout url when user logging out the current session. After logout it is redirecting to Azure logout screen and not to the given redirect url. Also its not clearing the cookies from the browser. 

So next time when user attempts to access the application it is directly taking to the application without asking login credentials and using credentials from cookies. One user can see other users details without logging in. It might leads to security issues.

Can someone help me to figure out this issue?

**Comment by rohitnarula7176** [comment]
@neelDev Can you specify which browser you are facing this issue on? Also can you share the logs fiddler trace for this use case.

**Comment by neelDev** [comment]
@rohitnarula7176 Thanks for your reply. I am facing this issue with almost all the browsers. I will update the fiddler trace shortly.

**Comment by neelDev** [comment]
@rohitnarula7176 Any updates on the above issue. For some reason fiddler is not working in my network. So unable to send the logs.

**Comment by rohitnarula7176** [comment]
@neelDev I will need some logs to see whats happening. If you can provide a small repro of your code, then I can debug it further,

**Comment by rohitnarula7176** [comment]
@neelDev Closing this issue for now. Please reopen if you still face issues.

**Comment by vishalwalke** [comment]
I am facing same Issue 
my Adal config is 


public get AdalConfig(): any {
        return {
            tenant: environment.tenant,
            clientId: environment.clientId, 
            redirectUri: https://localhost:4200,
            postLogoutRedirectUri: 'https://login.microsoftonline.com/<TenantId>/oauth2/logout?post_logout_redirect_uri=https://localhost:4200'
        };

**Comment by technicalganesh** [comment]
Can you guys please check the adalAuthenticationService's logOut method. look at the spelling, particularly it is uppercase O in logOut() instead of logout().

**Comment by benjaminao** [comment]
Hi, I have the same issue, can you help me with that, this is my Adal Configuration:

{
      tenant: 'XXXXXXXXXXXXXXXXXXXXXXXXXX',
      clientId: clientId,
      redirectUri: window.location.origin + '/#/home',
      postLogoutRedirectUri: window.location.origin + '/#/login',
      logOutUri: window.location.origin + '/#/login',
      extraQueryParameter: 'prompt=select_account'
}

When I call the **logout**() method go to a screen with this message 
"You signed out of your account
It's a good idea to close all browser windows."
and no redirect to my _postLogoutRedirectUri_ also when close and try to log in the user still **signed in.**

**Comment by ctientcheu** [comment]
any clear resolution to this issue?
Why is it closed when it is clearly still an issue?
it is happening in Chrome and firefox for me, I have not try any other
using latest version of adal-angular
calling logOut() and not logout()
I have tried all the above and more and nothing works... 

NOTE: After a successful logout (microsoft logout page is diplayed) and redirect to postLogoutRedirectUri, THE USER IS STILL SIGNED IN...

**Comment by benjaminao** [comment]
> @neelDev Closing this issue for now. Please reopen if you still face issues.

Hi, @neelDev may you help us with this problem? Why is it closed when it is clearly still an issue?
why you don't want a help us?

**Comment by olefrank** [comment]
Any news on this?

**Comment by duluong** [comment]
I got the same problem. 
adalSvc.logout() did not work.

**Comment by benjaminao** [comment]
Still waiting for a solution

**Comment by raviji** [comment]
Am also facing this same issue, please some one help me on this....

**Comment by andreav** [comment]
Same here.
* adal-angular 1.0.17

When i call  logOut I correctly get redirected to: 
    `https://<myadfsOnAzure>/adfs/oauth2/logout?post_logout_redirect_uri=https%3A%2F%2Flocalhost%3A8080%2F`

This url seems correct.
I found [someone](https://social.technet.microsoft.com/Forums/en-US/346b5ae1-c3d2-44e0-a713-977933d23ad7/oauth-logout-was-successful-but-the-client-will-not-be-redirected?forum=ADFS) saying the solution could be to include the "id_token_hint" as well. could this be the case?

I tryed this url and indeed it works redirecting me to my home page:

    `https://<myadfsOnAzure>/adfs/oauth2/logout?post_logout_redirect_uri=https%3A%2F%2Flocalhost%3A8080%2F&id_token_hint=<myidtoken>`
